lib/gems/pending/util/extensions/aws-sdk/s3_upload_stream_patch.rb

```diff
@@ -0,0 +1,281 @@
+# This is a backport of the `Aws::S3::Object#upload_stream` method that is
+# introduced in the V3 of the 'aws-sdk'
+#
+# To my knowledge, the changes from this patch should be completely backwards
+# compatible, and the API should easily support this change.
+#
+# Once upgraded, this file should be removed.
+
+require 'aws-sdk'
+
+if defined? Aws::S3::GEM_VERSION # This constant doesn't exist in v2
```

Comment:
So, I wouldn't want to push it just for this patch, but do we know if we are super far away from being able to use the new version as a whole? Would be curious to know how much effort that would be rather than forward porting this patch. @bronaghs Can you comment here or refer who would know about the aws-sdk version we are using?

Reply:
> do we know if we are super far away from being able to use the new version as a whole?

I don't know personally, or if it has been considered at all.

I want to say there was something that was being done to update `fog` in the recent future, and I thought that was potentially going to update the `aws-sdk`, but I might be remembering incorrectly (_**Update:**_  I was wrong... no surprise...  It was for https://github.com/ManageIQ/manageiq/pull/17258 and it was `fog-google`, not AWS... opps)

Reply:
Switching to v3 of the amazon API is not in the near future...  cc @bronaghs @agrare

Reply:
Looks like the far future has arrived? https://github.com/ManageIQ/manageiq-gems-pending/pull/374